### PR TITLE
vsphere: Reauthorize session when session is invalidated due to error

### LINKED ISSUE #330: vsphere CAA Reauthorize session in advent of an error
If a vcenter session is invalidated due to an error ... vcenter rebooting, etc the CAA should attempt to login and create a new authorized session.

### PR DESCRIPTION
**Pull request by caavery**
Fixes #330

Signed-off-by: Cathy Avery <cavery@redhat.com>

### CODE REVIEW

File: pkg/adaptor/hypervisor/vsphere/vsphere.go [hunk 1]
```diff
@@ -50,7 +53,7 @@ func NewGovmomiClient(vmcfg Config) (*govmomi.Client, error) {
 
 	us, _ := manager.UserSession(ctx)
 
-	logger.Printf("Vcenter api session %s created for user %s", us.Key, us.UserName)
+	logger.Printf("Vcenter api session key %s created for user %s", us.Key, us.UserName)
```
Comment (bpradipt) [inline pkg/adaptor/hypervisor/vsphere/vsphere.go:56]: Do we really need to log the key ?

File: pkg/adaptor/hypervisor/vsphere/vsphere.go [hunk 2]
```diff
@@ -76,12 +79,55 @@ func soapKeepAliveHandler(ctx context.Context, c *vim25.Client) func() error {
...
+	glock.Lock()
+
+	defer glock.Unlock()
+
+	active, err := gclient.SessionManager.SessionIsActive(ctx)
+
+	if active {
+		if err == nil {
+			us, _ := gclient.SessionManager.UserSession(ctx)
+			logger.Printf("Authenticated current session key %s for user %s", us.Key, us.UserName)
```
Comment (bpradipt) [inline pkg/adaptor/hypervisor/vsphere/vsphere.go:103]: Same question as https://github.com/confidential-containers/cloud-api-adaptor/pull/332/files#r1012495174

File: pkg/adaptor/hypervisor/vsphere/vsphere.go [hunk 3]
```diff
@@ -76,12 +79,55 @@ func soapKeepAliveHandler(ctx context.Context, c *vim25.Client) func() error {
...
+	err = gclient.SessionManager.Login(ctx, urlinfo.User)
+	if err != nil {
+		logger.Printf("Vcenter login failed error: %s", err)
+		return err
+	}
+
+	us, _ := gclient.SessionManager.UserSession(ctx)
+
+	logger.Printf("Created new authenticated session key %s for user %s", us.Key, us.UserName)
+
```
Comment (bpradipt) [inline pkg/adaptor/hypervisor/vsphere/vsphere.go:127]: same as the above comment - https://github.com/confidential-containers/cloud-api-adaptor/pull/332/files#r1012495174

Comment (caavery) [reply pkg/adaptor/hypervisor/vsphere/vsphere.go:127]: Nope we don't

File: pkg/adaptor/hypervisor/vsphere/vsphere.go (hunk 1 above)
Comment (caavery) [reply pkg/adaptor/hypervisor/vsphere/vsphere.go:56]: How about just the user?

**Review by bpradipt — APPROVED** [review]
/lgtm
Thanks @caavery

**Review by littlejawa — APPROVED** [review]
lgtm
Thanks!

**Pull request merged**